enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,255 @@
+
+### Drawbacks
+
+- N/A
+
+## Open Questions
+
+- Custom log paths: Can sos collect reports from dynamic paths? If not, can we symlink the log directory to as a means of faking a static path while writing logs to a custom location?  If so, can sos follow these symlinks?
+
+    - SOS can traverse symlinks to collect log files from a target directory.  The link itself and the target path are included in the tarball, meaning support can easily locate audit logs via the default location (`/var/log/kube-apiserver`).
+
+- Should Microshift support a `minFreeStorage` field that would be used to determine whether an acceptable amount of space exists at the log file path? This is not a value recognized by the apiserver, but it could be used by Microshift to check if the system is in an acceptable state to boot into.  If the free space is lower than the value, MicroShift would not start and log a fatal error to document the reason. If unset, MicroShift should deploy regardless of the available storage.
+
+    - Providing additional fields and internal handling logic is outside the scope of this EP. If this is a useful feature, it should be documented in a bespoke EP.
+
+## Test Plan
+
+* Unit tests to validate new config APIs
+* Integration tests to verify configured audit logging policies are applied properly
+
+## Graduation Criteria
+
+* Upstream code, tests, docs merged
+* Downstream builds, docs updated
+* Automated CI coverage for new functionality
+* QE test plans defined
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation completed and published
+- Sufficient test coverage
+- Gather feedback from users
+- Available by default
+
+### Tech Preview -> GA
+
+- Sufficient time for feedback
+- End-to-end tests
+
+### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+## Upgrade / Downgrade Strategy
+
+- The API must be backwards compatible for y-stream version
+
+## Operational Aspects of API Extensions
+
+- **Disk capacity exceeded**: MicroShift and the apiserver do not assess existing storage capacity for any logs, including audit logs.  As audit logs can grow quite quickly, this creates the potential for maxing out a storage device and hindering system performance.  Users must therefore consider their total storage needs, in addition to how and how often to transfer logs off-device.
+
+- **Invalid rotation values**: MicroShift will enforce a floor value of n>0 for `maxFiles`, `maxFileAge`, and `maxFileSize` fields.  Values of n <= 0 will be considered a configuration error and MicroShift will fail during startup. The invalid config field will be clearly logged along with the accepted value range.
```

Comment:
I could see some utility in allowing those fields to be 0 and only requiring that at least 1 is >0. For example, if I don't care how many files or what size are created but want logs kept for 30 days, I would say maxFiles=0, maxFileSize=0, and maxFileAge=30.

What requirements does the API server place on the values?

Reply:
I need to experiment with 0'ing the different values, as the documentation isn't clear.  The apiserver only checks if the values are below 0.  The only one with a documented 0 behavior is `maxFiles`, which means it will not limit the number of files retained.

Reply:
Okay, I've confirmed that the 0 value for maxFiles, maxFileAge, and MaxFileSize are all treated as disabling the the limit of the flag.

Given that, I think enforcing a min. of 1 flag with a value >0 wouldn't make sense and should be up to the user.  MicroShift should instead set reasonable defaults for nil values.